src/auto-trace.js

```diff
@@ -16,6 +16,10 @@ let globalMiddlewares = [];
  *   }))
  */
 export function catchError(callback, extraContext) {
+  if (callback instanceof Error) {
+    console.warn('catchError must be called before being passed to an onError handler.\nFind any instance of `catchError` and replace with `catchError()`.\nAlso check that an error object is not being passed into catchError as the first parameter should be a function.\n We call catchError to wrap the error with additional information, so the following error might be missing some additional useful context.')
+    throw callback
```

Comment:
Could we potentially take this one step further and use `catchError(callback)` so that instead of swallowing it, it is handled properly with the warning?

Reply:
This would still throw the error so it doesn't get swallowed. Before this change it would get swallowed though since this method returns a function and doesn't throw anything by itself

Reply:
oh sorry I meant catchError()(callback)